libraries/AP_WayBack/AP_WayBack.cpp

```diff
@@ -0,0 +1,818 @@
+/*
+
+this library collects points of drone position, records them in memory and eliminates 
+unnecessary data - redundant points and loops, so it always knows 
+the shortest way to home via visited points.
+ 
+ Copyright night_ghost@ykoctpa.ru 2017
+ 
+ license GPLV3 or above
+   
+*/
+
+#include "AP_WayBack.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_WayBack::Point* AP_WayBack::points;
+uint16_t          AP_WayBack::max_num_points=0; // size of storage
```

Comment:
>Lots of drones are Linux based

may be but I use 6g flight controller in 110g to 3kg drones with only 196k of memory, and definitely do this library for own needs too.

Check for multiple class instances will be added, but if there will be variant of hal.scheduler->register_io_process() that accepts AP_HAL::Proc and not only AP_HAL::MemberProc then class could be fully static.